dl1_data_handler/reader.py

```diff
@@ -140,13 +159,36 @@ def __init__(self,
                     mask = (img_ids != 0)
                     # TODO handle all selected channels
                     mask[mask] &= self._select_image(
-                        f.root[self.tel_type][img_ids[mask]]['charge'],
+                        f.root['/Images'][self.tel_type][img_ids[mask]]['charge'],
                         image_selection)
-                    for image_index, nrow in zip(img_ids[mask],
-                                               np.array(selected_nrows)[mask]):
-                        example_identifiers.append((filename, nrow,
-                                                    image_index, tel_id))
 
+                    mask[mask] &= self._select_image_from_file(
+                        f.root['/Images'][self.tel_type][img_ids[mask]]['charge'],
+                        parameters_table[img_ids[mask]],
+                        image_selection_from_file)
+
+                    if self.training_parameters == []:
+                        for image_index, nrow in zip(img_ids[mask], np.array(selected_nrows)[mask]):
+                            example_identifiers.append((filename, nrow,
+                                                        image_index, tel_id))
+                    else:
+                        param_list = []
+                        for parameter_name in self.training_parameters:
+                            if parameter_name != 'event_index':
+                                if parameter_name == 'hillas_log_intensity':
+                                    param = [np.log10(x['hillas_intensity']) for x in parameters_table if
+                                             x['hillas_intensity'] > -1]
+                                    param.insert(0, -1)
+                                    param_list.append(param)
+                                else:
+                                    param = [x[parameter_name] for x in parameters_table]
+                                    param_list.append(param)
+                        for image_index, nrow in zip(img_ids[mask], np.array(selected_nrows)[mask]):
+                            temp_list = []
+                            for list_index in range(len(param_list)):
+                                params_value = param_list[list_index][image_index]
+                                temp_list.append(params_value)
+                            example_identifiers.append((filename, nrow, image_index, tel_id, temp_list))
```

Comment:
The example_identifiers should only hold the identifiers and not any data. Therefore the parameters should merged in a list later i.e. def __getitem__(self, idx). Similar to the images. Also the transformation to log should go there.